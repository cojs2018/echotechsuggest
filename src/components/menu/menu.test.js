jest.mock('../newBookmark/newBookmark');
jest.mock('../manageBookmarks/manageBookmarks');

import React from 'react';
import render from 'react-test-renderer';
import { mount } from 'enzyme';

import NewBookmark from '../newBookmark/newBookmark';
import ManageBookmarks from '../manageBookmarks/manageBookmarks';
import Menu from './menu';

NewBookmark.mockReturnValue(<div><p>New Bookmark</p></div>);
ManageBookmarks.mockReturnValue(<div><p>Manage Bookmarks</p></div>);

describe('Menu', () => {
    it('renders correctly', () => {
        const tree = render
            .create(<Menu />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
});

test('Page handlers', async () => {
    let page = 0;
    const setPage = jest.fn(async (newPage) => page = newPage);

    React.useState = jest.fn()
        .mockReturnValueOnce([page, setPage]);

    const wrapper = mount(<Menu />);

    const createButton = wrapper.find('button').find('#create').at(0);
    createButton.simulate('click');

    await new Promise(resolve => setImmediate(resolve))
    expect(page).toStrictEqual(1);

    const manageButton = wrapper.find('button').find('#manage').at(0);
    manageButton.simulate('click');

    await new Promise(resolve => setImmediate(resolve))
    expect(page).toStrictEqual(2);
});