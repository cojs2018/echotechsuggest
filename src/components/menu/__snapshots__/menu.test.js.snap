// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1"
    >
      Echo-Tech-Suggest
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        id="create"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
            Create new bookmark
          </h3>
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        id="manage"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6z"
              />
              <path
                d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 10l-2.5-1.5L15 12V4h5v8z"
              />
            </svg>
            Manage bookmarks
          </h3>
        </span>
      </button>
    </div>
  </div>
</div>
`;
